diff -ur console_bridge-0.3.2_org/CMakeLists.txt console_bridge-0.3.2/CMakeLists.txt
--- console_bridge-0.3.2_org/CMakeLists.txt	2016-02-02 01:43:04.000000000 +0000
+++ console_bridge-0.3.2/CMakeLists.txt	2019-04-10 23:09:40.609135945 +0000
@@ -89,5 +89,5 @@
   ADDITIONAL_MAKE_CLEAN_FILES ${CMAKE_BINARY_DIR}/console_bridge-config.cmake
   ADDITIONAL_MAKE_CLEAN_FILES ${CMAKE_BINARY_DIR}/console_bridge.pc)
 
-enable_testing()
-add_subdirectory(test)
+#DISABLED enable_testing()
+#DISABLED add_subdirectory(test)
